tests/test_streaming.py

```diff
@@ -276,3 +276,109 @@ async def ret_a(x: str) -> str:  # pragma: no cover
+
+    @agent.tool
+    async def maybe_stop_run_tool(ctx: RunContext[int, tuple[str, int]], x: str) -> str:
+        if ctx.deps == 1:
+            ctx.stop_run(('abcdef', 777))
+        return f'deps={ctx.deps} x={x}'
+
+    # async with agent.run_stream('hello', deps=0) as result:
+    #     assert result.all_messages() == snapshot(
+    #         [
+    #             UserPrompt(content='hello', timestamp=IsNow(tz=timezone.utc)),
+    #             ModelStructuredResponse(
+    #                 calls=[ToolCall(tool_name='maybe_stop_run_tool', args=ArgsJson(args_json='{"x": "hello"}'))],
+    #                 timestamp=IsNow(tz=timezone.utc),
+    #             ),
+    #             ToolReturn(tool_name='maybe_stop_run_tool', content='deps=0 x=hello', timestamp=IsNow(tz=timezone.utc)),
+    #             ToolReturn(
+    #                 tool_name='final_result',
+    #                 content='Final result processed.',
+    #                 timestamp=IsNow(tz=timezone.utc),
+    #             ),
+    #         ]
+    #     )
+    #     assert await result.get_data() == snapshot(('deps=0 x=hello', 2))
+    #     assert result.all_messages() == snapshot(
+    #         [
+    #             UserPrompt(content='hello', timestamp=IsNow(tz=timezone.utc)),
+    #             ModelStructuredResponse(
+    #                 calls=[ToolCall(tool_name='maybe_stop_run_tool', args=ArgsJson(args_json='{"x": "hello"}'))],
+    #                 timestamp=IsNow(tz=timezone.utc),
+    #             ),
+    #             ToolReturn(tool_name='maybe_stop_run_tool', content='deps=0 x=hello', timestamp=IsNow(tz=timezone.utc)),
+    #             ToolReturn(
+    #                 tool_name='final_result',
+    #                 content='Final result processed.',
+    #                 timestamp=IsNow(tz=timezone.utc),
+    #             ),
+    #             ModelStructuredResponse(
+    #                 calls=[
+    #                     ToolCall(
+    #                         tool_name='final_result',
+    #                         args=ArgsJson(args_json='{"response": ["deps=0 x=hello", 2]}'),
+    #                     )
+    #                 ],
+    #                 timestamp=IsNow(tz=timezone.utc),
+    #             ),
+    #         ]
+    #     )
+
+    async with agent.run_stream('hello', deps=1) as result:
+        assert result.all_messages() == snapshot(
+            [
+                UserPrompt(content='hello', timestamp=IsNow(tz=timezone.utc)),
+                ModelStructuredResponse(
+                    calls=[ToolCall(tool_name='maybe_stop_run_tool', args=ArgsJson(args_json='{"x": "hello"}'))],
+                    timestamp=IsNow(tz=timezone.utc),
+                ),
+                ToolReturn(tool_name='final_result', content=('abcdef', 777), timestamp=IsNow(tz=timezone.utc)),
+            ]
+        )
+        assert await result.get_data() == snapshot(('abcdef', 777))
```

Comment:
I'm hitting:
```
E           pydantic_ai.exceptions.UnexpectedModelBehavior: Invalid message, unable to find tool. Called tool names: ['maybe_stop_run_tool']. Expected tool names: ['final_result']
```

And I can't for the life of me figure out what is happening or why. Could use some help understanding how streaming is supposed to work